stac_fastapi/api/stac_fastapi/api/app.py

```diff
@@ -234,3 +235,6 @@ def __attrs_post_init__(self):
 
         # customize openapi
         self.app.openapi = self.customize_openapi
+
+        # add gzip middleware
+        self.app.add_middleware(GZipMiddleware)
```

Comment:
Any reason to use `gzip` over `brotli`? We use brotli in titiler. Seemed to be both faster to compress and smaller sizes (at a medium compression level)
https://github.com/developmentseed/titiler/blob/47e3973a1fd26730b37b6e747d57733d35549848/titiler/application/titiler/application/main.py#L63

Reply:
It's built in with FastAPI is the only reason. Is it faster/smaller enough to be worth the extra dependency?

Reply:
Brotli adds a relatively small dependency and should make response transfer `faster` but I think we should avoid adding `client` dependencies. Adding custom Middlewares should be let to the users add the middleware of their choice.

Reply:
I don't have a really strong preference of gzip vs brotli here. Re @vincentsarago 's point, I figured this `stac_fastapi/api/app.py` was similar to titiler's `demo` that wouldn't be used by advanced users, but I'm not sure if that's the case?

Reply:
compression middleware switched to brotli